docs/design-v2.md

```diff
@@ -0,0 +1,132 @@
+and integrates with the pod scheduler to ensure a node with a replica mount is chosen on pod 
+failover.
+
+## Architecture and Components
+
+The diagram below shows the components in the Kubernetes control plane (CCP) and cluster 
+nodes, and the Azure services the Mandalay driver uses.
+
+![Mandalay Architecture](images/mandalay_arch.png)
+
+### Controller Plug-in
+
+In addition to the CSI Controller API Server, this plug-in hosts several controllers for 
+custom resources used to orchestrate persistent volume management and node placement.
+
+The controller plug-in is deployed as a [ReplicaSet](https://kubernetes.io/docs/concepts/
+workloads/controllers/replicaset/) through [Deployment](https://kubernetes.io/docs/concepts/
+workloads/controllers/deployment/) with leader election.
+
+### Node Plug-in
+
+In addition to the CSI Node API Server, this plug-in also provides feedback for pod placement 
+used by the scheduler extender described below.
+
+The node plug-in is deployed on each node in the cluster as a [DaemonSet](https://kubernetes.
+io/docs/concepts/workloads/controllers/daemonset/).
+
+### Scheduler Extender
+
+Mandalay provides a [scheduler extender](https://github.com/kubernetes/community/blob/master/
+contributors/design-proposals/scheduling/scheduler_extender.md) that is responsible for 
+influencing pod placements.
+
+Like the controller plug-in, the scheduler extender is deployed as a [ReplicaSet](https://
+kubernetes.io/docs/concepts/workloads/controllers/replicaset/) through [Deployment](https://
+kubernetes.io/docs/concepts/workloads/controllers/deployment/) with leader election.
+
+## Implementation
+
+This section describes the changes and high-level implementation details of the different 
+components in Mandalay.
+
+### StorageClass
+
+In addition to the existing StorageClass parameters described in [driver-parameters]
+(driver-parameters.md), Mandalay supports the following:
+
+| Name                                        | Meaning                                   | Available Value                                        | Mandatory | Default value                                                                                     |
+| ------------------------------------------- | ----------------------------------------- | ------------------------------------------------------ | --------- | ------------------------------------------------------------------------------------------------- |
+| `maxMountReplicaCount`                      | The number of mount replicas to maintain. | This value must be in the range `[0..(maxShares - 1)]` | No        | If `accessMode` is `ReadWriteMany`, the default is `0`. Otherwise, the default is `maxShares - 1` |
+| `enableFileSystemConsistencyCheck`          | ???                                       | `true` or `false`                                      | No        | `false`                                                                                           |
```

Comment:
The last two options are actually lower on the priority list. Do we want to share them right now?

Reply:
Nope.